docs/source/tutorials/memory_optimisations.rst

```diff
@@ -0,0 +1,243 @@
+All of our fine-tuning recipes support lower-precision optimizers from the `bitsandbytes <https://huggingface.co/docs/bitsandbytes/main/en/index>`_ library -
+a good place to start might be the ``Adam8bit`` and ``PagedAdamW`` optimizers, which we've tested our recipes with.
+
+To use this in your recipes, make sure you have installed bitsandbytes (``pip install bitsandbytes``). Then, enable
+a low precision optimizer using the :ref:`cli_label`:
+
+.. code-block:: bash
+
+  tune run <RECIPE> --config <CONFIG> \
+  optimizer._component_=bitsandbytes.optim.PagedAdamW
+
+or by directly :ref:`modify a config file<config_tutorial_label>`:
+
+.. code-block:: yaml
+
+  optimizer:
+    _component_: bitsandbytes.optim.PagedAdamW
+    lr: 2e-5
+
+.. note::
+
+  Utilising lower-precision optimizers as a memory-saving technique works best when using an optimizer which
+  maintains a state of gradient statistics. As an alternative, you could try a stateless optimizer
+  such as `stochastic gradient descent <https://pytorch.org/docs/stable/generated/torch.optim.SGD.html>`_
+  without momentum, which doesn't require any additional memory usage, and may result in lower memory usage
+  than a low-precision "stateful" optimizer. This concept is also important in :ref:`fusing the optimizer step
+  into the backward pass<glossary_opt_in_bwd>`.
+
+.. _glossary_opt_in_bwd:
+
+Fusing Optimizer Step into Backward Pass
+----------------------------------------
+
+We previously noted the distinction between stateful and stateless optimzers. As an alternative to lower-precision optimizers, let's
+consider a technique which enables the use of "stateful" optimizers such as ``AdamW``, without the memory overhead of gradient statistics,
+by completely removing the buffer of gradients which are stored by the optimizer during its ``step()``.
+
+We encourage you to read through the fantastic PyTorch tutorial on this concept:
+`How to save memory by fusing the optimizer step into the backward pass <https://pytorch.org/tutorials/intermediate/optimizer_step_in_backward_tutorial.html>`_.
+
+.. todo (SalmanMohammadi) ref full finetune
+
+You can enable this feature using the ``optimizer_in_bwd`` flag, which is currently only supported in our
+single-device full finetune recipe. This feature works best when gradient memory is particularly large
+i.e. when using a stateful optimizer with a model with a lot of parameters, and you don't need to use
+gradient accumulation.
+
+
+.. note::
+
+  You'll need to ensure you have PyTorch ``2.1.0`` or later to use this feature. See the PyTorch install instructions
+  `here <https://pytorch.org/get-started/locally/>`_.
+
+.. _glossary_peft:
+
+Parameter Efficient Fine-Tuning (PEFT)
+--------------------------------------
+
+.. _glossary_lora:
+
+Low Rank Adaptation (LoRA)
+^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Our tutorial on :ref:`finetuning Llama2 with LoRA<lora_finetune_label>` does a fantastic job of explaining LoRA, and how to use it. Here's
```

Comment:
I know you mean fantastic as in "the writer of this tutorial did a fantastic job", but reading it as a user it comes off as a bit self-congratulatory, so I would remove it (but regardless I do agree it is fantastic)

Reply:
Let the record show our appreciation for @ebsmothers